Resources/doc/first_steps.rst

```diff
@@ -249,6 +249,226 @@ to easily fetch objects based on multiple conditions:
+* The resolver will attempt to do a ``findOneBy()`` fetch by using
+  *all* of the wildcards in your route that are actually properties
+  on your document (non-properties are ignored).
+
+You can control this behavior by actually *adding* the ``MapDocument``
+attribute and using the `MapDocument options`_.
+
+Fetch via an Expression
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If automatic fetching doesn't work, you can write an expression using the
+`ExpressionLanguage component`_:
+
+.. code-block:: php
+
+    #[Route('/product/{product_id}')]
+    public function show(
+        #[MapDocument(expr: 'repository.find(product_id)')]
+        Product $product
+    ): Response {
+    }
+
+In the expression, the ``repository`` variable will be your document's
+Repository class and any route wildcards - like ``{product_id}`` are
+available as variables.
+
+This can also be used to help resolve multiple arguments:
+
+.. code-block:: php
+
+    #[Route('/product/{id}/comments/{comment_id}')]
+    public function show(
+        Product $product,
+        #[MapDocument(expr: 'repository.find(comment_id)')]
+        Comment $comment
+    ): Response {
+    }
+
+In the example above, the ``$product`` argument is handled automatically,
+but ``$comment`` is configured with the attribute since they cannot both follow
+the default convention.
+
+MapDocument Options
+~~~~~~~~~~~~~~~~~~~
+
+A number of options are available on the ``MapDocument`` attribute to
+control behavior:
+
+``id``
+    If an ``id`` option is configured and matches a route parameter, then
+    the resolver will find by the primary key:
+
+.. code-block:: php
+
+        #[Route('/product/{product_id}')]
+        public function show(
+            #[MapDocument(id: 'product_id')]
+            Product $product
+        ): Response {
+        }
+
+``mapping``
+    Configures the properties and values to use with the ``findOneBy()``
+    method: the key is the route placeholder name and the value is the Doctrine
+    property name:
+
+.. code-block:: php
+
+        #[Route('/product/{category}/{slug}/comments/{comment_slug}')]
+        public function show(
+            #[MapDocument(mapping: ['category' => 'category', 'slug' => 'slug'])]
+            Product $product,
+            #[MapDocument(mapping: ['comment_slug' => 'slug'])]
+            Comment $comment
+        ): Response {
+        }
+
+``exclude``
+    Configures the properties that should be used in the ``findOneBy()``
+    method by *excluding* one or more properties so that not *all* are used:
+
+.. code-block:: php
+
+        #[Route('/product/{slug}/{date}')]
+        public function show(
+            #[MapDocument(exclude: ['date'])]
+            Product $product,
+            \DateTime $date
+        ): Response {
+        }
+
+``stripNull``
+    If true, then when ``findOneBy()`` is used, any values that are
+    ``null`` will not be used for the query.
+
+``objectManager``
+    By default, the ``DocumentValueResolver`` uses the *default*
```

Comment:
Not sure how this works in ORM but I'd expect passing entity/document to a `ManagerRegistry` to figure out which object manager should be used by default?

Reply:
https://github.com/symfony/symfony/blob/503a7b3cb62fb6de70176b07bd1c4242e3addc5b/src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php#L82-L95 I think you mean this, if no parameter is passed, it does what you said.

Reply:
Cool, so the feature works as I've expected it's just the wording that confused me as `default` is often name of a default OM. How about

```suggestion
    By default, the ``DocumentValueResolver`` will choose DocumentManager that has the class registered for it
```

or maybe we have something better for when we're talking about multiple document managers? We rarely do (or at least did when I was working in such environment) so I'm not sure

Reply:
Changed to what you suggested, I haven't found any text in the docs, I think it is fine now and more understandable, thanks